Tasks/PowerShellOnTargetMachines2/task.json

```diff
@@ -114,16 +114,35 @@
             "helpMarkDown": "Arguments for the PowerShell script. Can be ordinal parameters or named parameters like -testParam test",
             "groupName": "ScriptOptions"
         },
+        {
+            "name": "InitializationScript",
+            "type": "string",
+            "label": "Initialization script",
+            "defaultValue": "",
+            "required": false,
+            "visibleRule": "ScriptType = FilePath",
+            "helpMarkDown": "Location of the data script for DSC on the target machines or on a UNC path like C:\\BudgetIT\\Web\\Deploy\\WebsiteConfiguration.ps1",
+            "groupName": "ScriptOptions"
+        },
+        {
+            "name": "SessionVariables",
+            "type": "multiLine",
+            "label": "Session Variables",
+            "defaultValue": "",
+            "required": false,
+            "visibleRule": "ScriptType = FilePath",
+            "helpMarkDown": "Set common session variables for both the scripts. For example, $variable = value, $var1 = \"value, 123\"",
```

Comment:
Should we instead take semi-colon separated list, so that we can just set this as is in session. We then don't have to interpret the input and this can be in native format as user would do in PS Shell.

Reply:
it makes things easier yes, but it also leaves room for people to use this input simply as a way to execute any arbitrary script as well (and if that is the case, then why not just specify it in the target script ? ). In the current way, we get to parse it first and then set variables accordingly.